app/src/vuex/modules/session.js

```diff
@@ -353,6 +358,45 @@ export default ({ apollo }) => {
     setNotificationAvailable(store, { notificationAvailable }) {
       state.notificationAvailable = notificationAvailable
     },
+    async getAllUsedAddresses(store) {
+      // filter local accounts to make sure they all have an address
+      const localAccounts = store.rootState.keystore.accounts.filter(
+        ({ address }) => address
+      )
+      // active sessions will likely overlap with the ones stored locally / in extension
+      const allAddresses = sortBy(
+        uniqWith(
+          localAccounts
+            .map((account) => ({
+              ...account,
+              networkId: account.network || account.networkId,
+              sessionType: `local`,
+            }))
+            .concat(
+              store.rootState.extension.accounts.map((account) => ({
+                ...account,
+                networkId: account.network || account.networkId,
+                sessionType: `extension`,
+              }))
+            )
+            .concat(state.allSessionAddresses),
```

Comment:
this should conceptually by "ledger addresses" + "explorer addresses", right? So we need all past addresses. session addresses is only the addresses that are used in any network right now. those addresses will disappear when you sign in with a different address for the same network.

Reply:
??

Reply:
I see addresses. They have no names though :shrug:

Reply:
I will change that in this PR too